### PR TITLE
Feature/server client hint ect memory

### PR DESCRIPTION
**Pull request by anton-karlovskiy**

### CODE REVIEW

File: functions/index.js [hunk 1]
```diff
@@ -95,6 +98,19 @@ app.get('/api/device', (req, res) => {
...
+  console.log('[server network-memory-considerate-model request] ECT, Device Memory => ', ect, deviceMemory);
+  
+  // TODO: As this is a demo, I think it should be easy enough to change these numbers as needed in the future. -> dotenv
+  const MEMORY_LIMIT = 4; // Threshold is 4GB RAM
+  const ECT_LIMIT = '4g';
+  // inspired by https://developers.google.com/web/fundamentals/performance/optimizing-content-efficiency/client-hints/#device_hints
+  const experienceType = (ect === ECT_LIMIT && deviceMemory > MEMORY_LIMIT) ? 'heavy' : 'light';
+  res.json({experienceType});
+});
+
```
Comment (addyosmani) [inline functions/index.js]: I see that we are applying this logic in two places - functions/index.js and then again in node-ect-memory-considerate-loading/functions/index.js. I may have missed something earlier, but can you describe why these are in two different places?

File: node-ect-memory-considerate-loading/server.js [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+  const deviceMemory = req.headers['device-memory'];
+  console.log('[server network-memory-considerate-model request] ECT, Device Memory => ', ect, deviceMemory);
+  
+  // TODO: As this is a demo, I think it should be easy enough to change these numbers as needed in the future. -> dotenv
+  const MEMORY_LIMIT = 4; // Threshold is 4GB RAM
+  const ECT_LIMIT = '4g';
+  // inspired by https://developers.google.com/web/fundamentals/performance/optimizing-content-efficiency/client-hints/#device_hints
+  const experienceType = (ect === ECT_LIMIT && deviceMemory > MEMORY_LIMIT) ? 'heavy' : 'light';
+  res.json({experienceType});
+});
```
Comment (addyosmani) [inline node-ect-memory-considerate-loading/server.js]: Similar to my earlier comment, I feel a little confused why we have three separate places where Express routing + a check for memory and ECT are required (they all appear to be doing very similar work). Could you summarize?

File: functions/index.js (hunk 1 above)
Comment (anton-karlovskiy) [reply functions/index.js]: Sorry for confusion. node-ect-memory-considerate-loading/functions is any longer useless. It's just for firebase testing focusing on that project. In future, we can just use the functions/index.js. I'll remove subproject/functions for all PRs. And also after merge, I'll double check the whole project again.

File: node-ect-memory-considerate-loading/server.js (hunk 2 above)
Comment (anton-karlovskiy) [reply node-ect-memory-considerate-loading/server.js]: Yes, please check the above reply. Thanks.

**Comment by anton-karlovskiy** [comment]
@addyosmani 
The conflicts are resolved.
